ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/Benchmark.scala

```diff
@@ -41,87 +35,92 @@ object Benchmark {
     Future
       .traverse(streamConfigs) {
         case streamConfig: StreamConfig.TransactionsStreamConfig =>
-          StreamMetrics
-            .observer[GetTransactionsResponse](
-              streamName = streamConfig.name,
-              logInterval = reportingPeriod,
-              metrics = MetricsSet.transactionMetrics(streamConfig.objectives),
-              logger = logger,
-              exposedMetrics = Some(
-                MetricsSet
-                  .transactionExposedMetrics(streamConfig.name, metricRegistry, reportingPeriod)
-              ),
-              itemCountingFunction = MetricsSet.countFlatTransactionsEvents,
-              maxItemCount = streamConfig.maxItemCount,
-            )(system, ec)
-            .flatMap { observer =>
-              streamConfig.timeoutInSecondsO
-                .foreach(timeout => scheduleCancelStreamTask(timeout, observer))
-              apiServices.transactionService.transactions(streamConfig, observer)
-            }
+          for {
+            _ <- delaySubscriptionIfConfigured(streamConfig)(system)
```

Comment:
You're adding the delay in a few places here. Could you instead add it once in `com.daml.ledger.api.benchtool.LedgerApiBenchTool#benchmarkStreams` which is the only call site of `Benchmark.run`?

Reply:
Oh, you're delaying each stream separately which is more powerful and straightforward to do